### DISCUSSION TITLE
Gulpfile presets

### DISCUSSION THREAD

**Discussion by binyamin** [gulpjs/gulp#2650] [Help] [answered]
I find that my Gulp tasks are often pretty similar across projects. this makes it tedious to start a new project, because I need to find the most up-to-date version of my gulp tasks. What if you could just install a preset of tasks, which you could update as needed?

Truth is, this doesn't _need_ to be hard-coded. You could also export a bunch of gulp tasks as a module, so you could install and use it anywhere. I do think, though, that there should be a standardized way of doing this, for compatibility.

**Comment by phated** [comment, marked as answer]
@binyamin this is solved by Undertaker custom registries. We actually had an awesome contributor add those docs to the website: https://gulpjs.com/docs/en/advanced/creating-custom-registries

**Reply by binyamin** [reply]
That's pretty technical, and requires knowledge of gulp's inner workings. Why would I use a registry, instead of exporting a bunch of predefined tasks?

**Reply by phated** [reply]
Did you even read the page? https://gulpjs.com/docs/en/advanced/creating-custom-registries/#sharing-tasks
